src/Fmt_ast.ml

```diff
@@ -1077,9 +1077,21 @@ and fmt_expression c ?(box= true) ?epi ?eol ?parens ?ext
         $ fmt_if_k (not last) (break_unless_newline 1 0)
       in
       let is_nested_diff_prec_infix_ops =
+        let skip_parent_for_some_infix x =
+          match x.pexp_desc with
+          | Pexp_ident {txt= Lident i} -> (
+            match i with
+            | "==" | "=" | "<" | ">" | "<=" | ">=" | "<>" | "!=" -> true
```

Comment:
I'm a little wary of what seems like hard-coding a special case. Is there a more general rule that this is implementing, that could be used as an explanation for which infix ops get parenthesized and which don't? I wonder if it would make sense to omit parens for (assignment and) infix ops of precedence InfixOp0 (that is, ["!="] and symbols starting with ['=' | '<' | '>' | '|' | '&' | '$'])? I guess the question is whether it is really just these ops that you think shouldn't have parens, or whether these are just the only ones of the low precedence which are defined in the default environment?

Probably best would be to omit parens in the case where the type of the infix op arg differs from that of the return type, but the formatter (currently) operates on the untyped tree.